Remove Vault Connection from the manager.

revit13 opened the pull request:
This PR removes Vault connection from the manager as it is currently not used in the manager code.

shlomitk1 commented on manager/controllers/utils/configuration.go:
```diff
@@ -15,9 +15,16 @@ import (
+	// The manager role should be passed as env var after it was configured
+	// with the appropriate permissions for the manager.
+	VaultSecretKey   string = "VAULT_TOKEN"
+	VaultTTLKey      string = "VAULT_TTL"
```
We are not using neither VAULT_TOKEN nor VAULT_TTL, so why keep it?

shlomitk1 replied:
Not sure what is the purpose of this PR. 
1. manager role env is not introduced, only mentioned in a comment
2. If vault token should be obtained by the manager, why is it still in the configmap and configuration.go file?

revit13 commented:
Following a private talk with @shlomitk1 I changed the pr to remove Vault connection from the manager code and updated the pr description accordently.

shlomitk1 commented:
@revit13 Please remove VAULT_TTL and VAULT_TOKEN from the chart values/ m4d-config as well.

roee88 commented:
@revit13 is there an issue for changing to make the manager setup policies dynamically?

revit13 commented:
@shlomitk1 Thanks will change that. @roee88 in issue  #585 the last task refers to that.

revit13 commented:
@shlomitk1 I left charts/m4d/templates/vault-credentials.yaml as it is still used...

shlomitk1 commented:
> 
> 
> @shlomitk1 I left charts/m4d/templates/vault-credentials.yaml as it is still used...

Secret should be left, but the reference to it in the manager deployment is no longer required.
https://github.com/mesh-for-data/mesh-for-data/blob/f06f70e3c6a3dcf643d7416a8578fba4f14a50ae/charts/m4d/templates/manager-deployment.yaml#L81-L84

revit13 commented:
> 
> 
> > @shlomitk1 I left charts/m4d/templates/vault-credentials.yaml as it is still used...
> 
> Secret should be left, but the reference to it in the manager deployment is no longer required.
> 
> https://github.com/mesh-for-data/mesh-for-data/blob/f06f70e3c6a3dcf643d7416a8578fba4f14a50ae/charts/m4d/templates/manager-deployment.yaml#L81-L84

Thanks @shlomitk1 Fixed.

roee88 commented on pkg/vault/vault_interface_test.go:
```diff
@@ -22,10 +22,11 @@ func Log(t *testing.T, label string, err error) {
 	os.Setenv("RUN_WITHOUT_VAULT", "1")
-	t.Logf("Token = " + utils.GetVaultToken())
-	conn, err := InitConnection(utils.GetVaultAddress(), utils.GetVaultToken())
+	t.Logf("Token = " + vaultToken)
```
No tokens in logs

revit13 replied:
Fixed Thanks
